### ISSUE TITLE
Colour of dependent points

### ISSUE THREAD

**Issue by kranich** [CindyJS/CindyJS#217]
Dependent points are drawn in darker colours in Cinderella but not in CindyJS.

I am pretty sure this has been mentioned somewhere already (maybe on the mailing list) but I could not find a duplicate issue.

**Comment by gagern** [comment]
CindyJS already has `defaultAppearance.dimDependent` which [darkens colors of non-`Free` points](https://github.com/CindyJS/CindyJS/blob/f0f53e6fe91472ea30cfce666d8640992c7df7f3/src/js/libgeo/GeoBasics.js#L67). This really makes an exception only for `Free`, while the exception should be linked to the `isMovable` property of the operator (or the `movable` property of the element, which is the same), in combination with the `pinned` property of the element.

The setting [defaults to 1](https://github.com/CindyJS/CindyJS/blob/f0f53e6fe91472ea30cfce666d8640992c7df7f3/src/js/libgeo/GeoBasics.js#L10) and currently Cinderella doesn't export its value. I just now found that Cinderella by default uses a dim factor of 0.7. So we should check the `darkenDependent` attribute (which in the source code appears to be tied to the port) and set the appearance accordingly.

**Comment by gagern** [comment]
Urgh! I just realized that things are more complicated than I anticipated. Cinderella does darken dependent points only if their color is chosen from the palette. For colors defined by a color vector, e.g. using a script line like `P.color = [1,1,0]`, the color is *not* dimmed even if the point in question is dependent. So how do we deal with this?

1. Export the final color for every point, and either leave `defaultAppearance.dimDependent = 1` by default or explicitly set that value when exporting.
2. Export a per-element `dimDependent` setting, and explicitly set this to `false` if the global default is `true` but a given point has a non-palette color.
3. Export the palette index and do the same palette magic in CindyJS as well. Required if we want to support the `color` inspectable property, since that represents a palette index.

If we go with 2., then #231 may need some extension to support that. And all three require some work on the Cinderella side. Note that a non-palette color currently becomes red on export, which is certainly incorrect.
